Feature/home screen design

abdo-essam opened the pull request:
<img width="1280" height="2856" alt="image" src="https://github.com/user-attachments/assets/de706fe2-5052-4841-83dc-14a160b98401" />

abdo-essam committed:
feat(ui): Implement home screen design

Adds header, categories, products sections.

abdo-essam committed:
Merge remote-tracking branch 'origin/develop' into feature/home-screen-design

abdo-essam committed:
Merge remote-tracking branch 'origin/develop' into feature/home-screen-design

MohamedOmar989 committed:
Merge remote-tracking branch 'origin/develop' into feature/Category-Tabs-and-Products-Grid

israamohamed123 committed:
Merge branch 'develop' into feature/home-screen-design

abdo-essam committed:
feat(ui): Add special offers section to home

Adds special offer card and section to the home screen.
Includes image asset and model for special offers.

abdo-essam committed:
Merge remote-tracking branch 'origin/develop' into feature/home-screen-design

abdo-essam committed:
refactor(ui): Refactor special offers section

Refactors special offers section for improved code structure
and removed unused imports. Updates section header to use
transparent background.

MohamedOmar989 committed:
Merge remote-tracking branch 'origin/develop' into feature/Category-Tabs-and-Products-Grid

israamohamed123 committed:
Merge remote-tracking branch 'origin/feature/home-screen-design' into feature/home-screen-header-section

MohamedOmar989 committed:
Merge remote-tracking branch 'origin/feature/home-screen-design' into feature/Category-Tabs-and-Products-Grid

MohamedOmar989 committed:
feat: implement category tabs and update chip

MohamedOmar989 committed:
feat: Implement favorite functionality and products list

MohamedOmar989 committed:
refactor: extract widgets from HomeScreen

MohamedOmar989 committed:
feat: add category icons

MohamedOmar989 committed:
feat: add new category and favorite icons

abdo-essam committed:
Merge remote-tracking branch 'origin/develop' into feature/home-screen-design

israamohamed123 committed:
header section and search bar

israamohamed123 committed:
Merge remote-tracking branch 'origin/feature/home-screen-design' into feature/home-screen-header-section

israamohamed123 committed:
remove gap library

abdo-essam committed:
Merge remote-tracking branch 'origin/develop' into feature/home-screen-design

M-Abdelnabi committed:
Home screen Top Stores Section (#41)

* fix favourite icon in store card

* add discount tag widget and its icons

* use discount tag with store card

* refactor discount tag widget

* rename icon tag

* refactor discount tag inside the store card

* use nested containers to simplify the usage of padding

* use nested containers to simplify the usage of padding

* make the discount tag optional

* add card click option

* add card click option

* add top stores section

* handle card clicks

* remove unnecessary padding

* remove unnecessary padding from section header

* refactor store card

* feat(ui): Add top stores section to home screen

Adds TopStoresSection widget to display a list of stores
with images, names, and discounts. Includes StoreCard
widget and supporting assets.

---------

Co-authored-by: abdo-essam <abdo.e.mosa@gmail.com>

abdo-essam committed:
Merge branch 'refs/heads/feature/home-screen-design' into feature/Category-Tabs-and-Products-Grid

# Conflicts:
#	lib/core/design_system/constants/assets.dart
#	lib/ui/screens/home/home_screen.dart

abdo-essam committed:
refactor(ui): Refactor home screen and widgets

Move chip category and refactor home screen widgets for better organization and maintainability.

abdo-essam committed:
feat(ui): Implement category tabs and product grid

Implements the category tabs and product grid on the
home screen. Includes product cards and images.

abdo-essam committed:
Merge remote-tracking branch 'origin/develop' into feature/home-screen-design

abdo-essam committed:
Merge branch 'feature/home-screen-design' into feature/home-screen-header-section

# Conflicts:
#	lib/ui/screens/home/home_screen.dart

abdo-essam committed:
refactor(ui): Refactor home screen

Refactored the home screen to use sections.
Improved code organization and readability.
Moved products list to product section.

abdo-essam committed:
fix(ui): Fix carriage return

Remove carriage return characters from home_screen.dart

abdo-essam committed:
fix(ui): Navigate to MainScreen on app start

Switches initial route from HomeScreen to MainScreen.

abdo-essam committed:
feat(ui): Implement home screen data

Adds DataProvider and restructures home screen

abdo-essam committed:
fix(ui): Add missing special offer

Added a missing special offer to the list.
Also, fixed formatting in DataProvider.dart

abdo-essam committed:
refactor(ui): Reformat main.dart

Fixed formatting issues in main.dart.

abdo-essam committed:
Merge remote-tracking branch 'origin/develop' into feature/home-screen-design

ghost commented:
<!-- sellio-bot-autoupdate-comment -->
**[Sellio Bot] Auto-update Status**


✅ Branch successfully updated from `develop`.

ghost commented:
<!-- sellio-bot-autoupdate-comment -->
**[Sellio Bot] Auto-update Status**


✅ Branch successfully updated from `develop`.

abdo-essam committed:
chore(config): Update flutter_gap dependency

Update flutter_gap dependency in pubspec.yaml.
This adds flutter_gap: ^1.2.0, replacingchore(config): Update flutter_gap version

Update flutter_gap dependency in pubspec.yaml.
 the previous
flutter_svg line. No functional change expected.

ghost commented:
<!-- sellio-bot-autoupdate-comment -->
**[Sellio Bot] Auto-update Status**


✅ Branch successfully updated from `develop`.

ghost commented:
<!-- sellio-bot-autoupdate-comment -->
**[Sellio Bot] Auto-update Status**


✅ Branch successfully updated from `develop`.
